docs/userguide/utils.md

```diff
@@ -0,0 +1,128 @@
+useful - for instance if your estimate already is several Exabytes, it is unlikely that
+most computers would be able to handle the result even if there are constraints present.
+```
+
+```{admonition} Influence of Continuous Parameters
+:class: info
+Continuous parameters fo not influence the size of the discrete search space part.
+Hence, they are ignored by the utility.
+```
+
+```{admonition} Memory During Optimization
+:class: warning
+`estimate_product_space_size` only estimates the memory required to handle the search
+space. **It does not estimate the memory required during optimization**, which can be
+of a similar magnitude, but generally depends on additional factors.
+```
+
+```{admonition} Effective Search Space Creation for Mixtures
+:class: tip
+If you run into issues creating large search spaces, as for instance for mixtures, you
+can try to use the [`SubspaceDiscrete.from_simplex`](baybe.searchspace.discrete.SubspaceDiscrete.from_simplex)
+constructor. Instead of creating the search space completely before filtering it down
+according to the constraints, this constructor includes the main mixture constraint
+already during the Cartesian product, requiring substantially less memory overall. In
+addition, BayBE can also be installed with an optional `polars` dependency (`pip install
+baybe[polars]`) that will utilize the more efficient machinery form polars for handling
+of the search space and its constraints.
+```
+
+## Reproducibility
+In some scenarios, for instance when testing your code setup, it can be useful to fix
+the random seeds for all relevant engines to generate reproducible results. BayBE offers
+the [`set_random_seed`](baybe.utils.random.set_random_seed) utility for this purpose:
+
+```python
+from baybe.utils.random import set_random_seed
+
+# Set the global random seed for all relevant engines
+set_random_seed(1337)
+
+# Assuming we have a prepared campaign
+campaign.recommend(5)
+```
+
+Setting the global random seed can be undesirable if there are other packages in your
+setup. For this, BayBE offers [`temporary_seed`](baybe.utils.random.temporary_seed):
+
+```python
+from baybe.utils.random import temporary_seed
+
+# Set the random seed for all relevant engines temporarily within the context
+with temporary_seed(1337):
+    campaign.recommend(5)
+```
+
+## Add Fake Target Measurements and Noise
+When creating test scripts, it is often useful to try the recommendation loop for a few
+iterations. However, this requires some arbitrary target measurements to be set. Instead
+of coming up with a custom logic every time, you can use the
+[`add_fake_results`](baybe.utils.dataframe.add_fake_results) utility to add fake target
+measurements and the [`add_parameter_noise`](baybe.utils.dataframe.add_parameter_noise)
+utility to add artificial parameter noise:
+
+```python
+from baybe.utils.dataframe import add_fake_results, add_parameter_noise
```

Comment:
One other thing: do we want to finally fix the inplace behavior (i.e. have we already talked about that?). I think we should either make the inplace an optional flag or at least always also return the modified dataframe. Thoughts welcome

Reply:
so remind me again, this was achieved by modifying inplace + returning the modified object right? Dont see much reason against that and can add it

Reply:
changed